### PR TITLE
Refine filters - PST-122, PST-138

### PR DESCRIPTION
**Pull request by AlbinaMuzafarova**
- clickaway closes open filter chips or list picker
- filter removed if empty after update (_note_: filter is added on chip click by design)
- filter shown (red) during edit
- selected terms left-aligned
- add button aligned with filters

also includes changes authored by @samuelhuebner in `fix/filter-item-alignments`

### CODE REVIEW

File: client/src/components/Filters.vue
Comment (micahh2) [inline client/src/components/Filters.vue:3]: If you enter an invalid date and then click away, you lose the "add date" option forever.
image

**Review by micahh2 — APPROVED** [review]
Great work @AlbinaMuzafarova . I know all of these filters are finicky and difficult to work with (which I take most of the blame for).
I don't know if you're looking for general notes, but, in general we should be looking for ways to make components less complicated and move that complexity elsewhere (e.g. the store, or service helpers).

As usual though, good Job. :+1:

**Review by icicala — COMMENTED** [review]
@AlbinaMuzafarova I am getting strange behaviour. On adding filter the whole page is getting out of context how you can seen in the image below. 
image

**Comment by AlbinaMuzafarova** [comment]
thank you all for your reviews! i'll close the PR to rework on the indicated issues.

File: client/src/components/Filters.vue
Comment (AlbinaMuzafarova) [reply client/src/components/Filters.vue]: @micahh2 thank you for noting this! I'm not sure I changed anything significant to handle this case but i'm having difficulties replicating the behavior. I tried with letters and two digits only - seems to show the chip on my side. Maybe I haven't considered the edge case you tried. Do you remember what kind of invalid date you used?

**Comment by AlbinaMuzafarova** [comment]
@icicala I think the issue you pointed out was there before my changes. I think it's related to the changing size of the menu options. Will see if I can fix it quickly within the scope of the ticket

**Comment by icicala** [comment]
> @icicala I think the issue you pointed out was there before my changes. I think it's related to the changing size of the menu options. Will see if I can fix it quickly within the scope of the ticket

You are right! I reproduced now without your changes. Sorry I have noticed it before. Danke @AlbinaMuzafarova for investigation. Well Done!

**Comment by icicala** [comment]
> > @icicala I think the issue you pointed out was there before my changes. I think it's related to the changing size of the menu options. Will see if I can fix it quickly within the scope of the ticket
> 
> You are right! I reproduced now without your changes. Sorry I have not noticed it before. Danke @AlbinaMuzafarova for investigation. Well Done!

File: client/src/components/Filters.vue
Comment (micahh2) [reply client/src/components/Filters.vue]: @AlbinaMuzafarova Sorry I didn't provide more detailed steps!
1) Open Options Menu & Click add 
2) Click "+ date"
2) Enter 123 and 123 for the two dates
4) Click away (or let edit timeout)

Oddly enough it sometimes shows up again once the page is reloaded, but only if you've first added another "empty" filter.

Comment (micahh2) [reply client/src/components/Filters.vue]: I looks like on the latest version of your branch this behavior is fixed. :+1:

Comment (AlbinaMuzafarova) [reply client/src/components/Filters.vue]: thank you Micah, that helps a lot! 

I will keep this thread open until next feedback, although it seems to behave alright now.

**Review by icicala — APPROVED** [review]
It works fine for me. Danke @AlbinaMuzafarova for your contribution.

**Review by samuelhuebner — APPROVED** [review]
Finally the filters are more usable 👍🏼 🔥 

A very small detail that occured to me:
The date input dropdown alignment is a little bit off. And also the year is a little bit cut off. 
image

**Comment by AlbinaMuzafarova** [comment]
@samuelhuebner should be fixed now

**Review by duaakelzi — APPROVED** [review]
works fine thank you Albina

**Comment by AlbinaMuzafarova** [comment]
to reduce merge conflicts will merge this branch once PR 88 approved

**Pull request merged**
